homeassistant/components/fyta/sensor.py

```diff
@@ -0,0 +1,178 @@
+        device_class=SensorDeviceClass.ENUM,
+        options=PLANT_STATUS_LIST,
+        value_fn=lambda value: PLANT_STATUS[value],
+    ),
+    FytaSensorEntityDescription(
+        key="moisture_status",
+        translation_key="moisture_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=PLANT_STATUS_LIST,
+        value_fn=lambda value: PLANT_STATUS[value],
+    ),
+    FytaSensorEntityDescription(
+        key="salinity_status",
+        translation_key="salinity_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=PLANT_STATUS_LIST,
+        value_fn=lambda value: PLANT_STATUS[value],
+    ),
+    FytaSensorEntityDescription(
+        key="temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    FytaSensorEntityDescription(
+        key="light",
+        translation_key="light",
+        native_unit_of_measurement="mol/d",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    FytaSensorEntityDescription(
+        key="moisture",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.MOISTURE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    FytaSensorEntityDescription(
+        key="salinity",
+        translation_key="salinity",
+        native_unit_of_measurement="mS/cm",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    FytaSensorEntityDescription(
+        key="ph",
+        device_class=SensorDeviceClass.PH,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    FytaSensorEntityDescription(
+        key="battery_level",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+]
+
+PLANT_NUMBER_SENSOR: FytaSensorEntityDescription = FytaSensorEntityDescription(
+    key="plant_number",
+    translation_key="plant_number",
+    state_class=SensorStateClass.MEASUREMENT,
+    entity_category=EntityCategory.DIAGNOSTIC,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the FYTA sensors."""
+    coordinator: FytaCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    plant_entities: list[CoordinatorEntity] = [
+        FytaCoordinatorSensor(coordinator, entry, sensor)
+        for sensor in SENSORS
+        if sensor.key in coordinator.data
+    ]
+    plant_entities.append(
+        FytaCoordinatorSensor(coordinator, entry, PLANT_NUMBER_SENSOR)
+    )
+
+    plant_entities.extend(
+        [
+            FytaPlantSensor(coordinator, entry, sensor, plant_id)
+            for plant_id in coordinator.fyta.plant_list
+            for sensor in SENSORS
+            if sensor.key in coordinator.data[plant_id]
+        ]
+    )
```

Comment:
```suggestion
    plant_entities.extend(
            FytaPlantSensor(coordinator, entry, sensor, plant_id)
            for plant_id in coordinator.fyta.plant_list
            for sensor in SENSORS
            if sensor.key in coordinator.data[plant_id]
    )
```

I _think_ this also works, it doesn't create a new list so its cheaper